osu.Game.Rulesets.Taiko/UI/DrumTouchInputArea.cs

```diff
@@ -163,23 +216,44 @@ protected override void PopOut()
             mainContent.FadeOut(300);
         }
 
-        private partial class QuarterCircle : CompositeDrawable, IKeyBindingHandler<TaikoAction>
+        private partial class DrumSegment : CompositeDrawable, IKeyBindingHandler<TaikoAction>
         {
-            private readonly Circle overlay;
+            private TaikoAction action;
+
+            public TaikoAction Action
+            {
+                get => action;
+                set
+                {
+                    if (action == value)
+                        return;
+
+                    action = value;
+                    updateColoursFromAction();
+                }
+            }
+
+            private Circle overlay = null!;
 
-            private readonly TaikoAction handledAction;
+            private Circle circle = null!;
 
-            private readonly Circle circle;
+            [Resolved]
+            private OsuColour colours { get; set; } = null!;
 
             public override bool Contains(Vector2 screenSpacePos) => circle.Contains(screenSpacePos);
 
-            public QuarterCircle(TaikoAction handledAction, Color4 colour)
+            public DrumSegment(TaikoAction action)
```

Comment:
```suggestion
            public DrumSegment()
```

Since the correct action will be set by `configTouchControlScheme.BindValueChanged(..., true)` in `DrumTouchInputArea.load()`, having this in the ctor is misleading.

Reply:
Oops, meant to do this.